src/tarantool.client/NetworkStreamPhysicalConnection.cs

```diff
@@ -23,8 +24,18 @@ internal class NetworkStreamPhysicalConnection : INetworkStreamPhysicalConnectio
 
         public void Dispose()
         {
-            _disposed = true;
-            _stream?.Dispose();
+            lock (this)
+            {
+                if (_disposed)
+                {
+                    return;
+                }
+
+                _disposed = true;
+
+                _stream?.Dispose();
+                _socket?.Dispose();
```

Comment:
Stream owns socket and will dispose it, because we pass True to stream parameter.